src/renderer_d3d12.cpp

```diff
@@ -6422,6 +6422,12 @@ namespace bgfx { namespace d3d12
 				bool constantsChanged = draw.m_uniformBegin < draw.m_uniformEnd;
 				rendererUpdateUniforms(this, _render->m_uniformBuffer[draw.m_uniformIdx], draw.m_uniformBegin, draw.m_uniformEnd);
 
+				// The state reset on D3D12 is skipped because draw.m_streamMask is set to 0 when the current draw.m_scissor is not set ( equal to UINT16_MAX )
```

Comment:
m_streamMask is set to 0 only for dummy draw calls like `bgfx::touch`.

Reply:
Also if this was fix for D3D12, why completely different fix for VK when the code is almost identical?

Reply:
Vulkan doesn't have the concept of Stream Mask, so it's far from being almost identical. If this was the case, moving the code on top would fix it, but indeed causes another series of glitches graphically :)

Reply:
> Vulkan doesn't have the concept of Stream Mask

https://github.com/bkaradzic/bgfx/pull/2276/commits/4facc9560410cc0e01600ea372dacad4fb843a22#diff-4fe4dacbf31bf95719870316660d20e1deca6ebbfa3f39c6b2b8b0b302d629f9R6930-R6932

Reply:
You're right my bad. I forgot about that. Then no idea honestly, perhaps there's something missing in the D3D12 code. Anyway I've tested this patch intensively in my own code flow and works fine as expected. As well as in your demos. Tbh I'd merge this anyway and improve it later if a better solution is found.

Reply:
It would be the best if you recreate simplest possible repro case for this issue, and I'll fix it. Repro case you submitted before is not clear. Try with 01-cubes example, and only add code that's necessary to reproduce the bug.